bigchaindb/backend/mongodb/query.py

```diff
@@ -70,33 +79,83 @@ def get_blocks_status_from_transaction(conn, transaction_id):
 
 @register_query(MongoDBConnection)
 def get_txids_by_asset_id(conn, asset_id):
-    return conn.db['bigchain']\
-            .find({'block.transactions.asset.id': asset_id},
-                  projection=['id'])
+    # get the txid of the create transaction for asset_id
+    cursor = conn.db['bigchain'].aggregate([
+        {'$match': {
+            'block.transactions.id': asset_id,
+            'block.transactions.operation': 'CREATE'
+        }},
+        {'$unwind': '$block.transactions'},
+        {'$match': {
+            'block.transactions.id': asset_id,
+            'block.transactions.operation': 'CREATE'
```

Comment:
This is only to retrieve the the create transaction. The transfer is done in a separate query